python-package/lightgbm/basic.py

```diff
@@ -2250,6 +2254,10 @@ def get_data(self):
                     self.data = self.data.iloc[self.used_indices].copy()
                 elif isinstance(self.data, dt_DataTable):
                     self.data = self.data[self.used_indices, :]
+                elif isinstance(self.data, Sequence):
+                    self.data = self.data[self.used_indices]
+                elif isinstance(self.data, list) and len(self.data) > 0 and all(isinstance(x, Sequence) for x in self.data):
+                    self.data = np.array([row for row in self._yield_row_from_seqlist(self.data, self.used_indices)])
```

Comment:
Please add test cases with `Sequence` and `List[Sequence]` for `add_features_from()` here
https://github.com/microsoft/LightGBM/blob/90342e929b1801fb2a20b410e2db5a69a9063b48/tests/python_package_test/test_basic.py#L325

Reply:
Also, please enhance this test (or just add a new one)
https://github.com/microsoft/LightGBM/blob/d05f54701ee88a1373bd9e0a1ea4317827015c9c/tests/python_package_test/test_basic.py#L133
with new `get_data()` functionality for `Sequence(s)` raw data.

Reply:
Thanks for the review. I'll add test cases later this week.